tests/components/husqvarna_automower/test_button.py

```diff
@@ -92,6 +93,55 @@ async def test_button_states_and_commands(
         )
 
 
+@pytest.mark.freeze_time(datetime.datetime(2024, 2, 29, 11, tzinfo=datetime.UTC))
+async def test_sync_clock(
+    hass: HomeAssistant,
+    mock_automower_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test button commands."""
+    entity_id = "button.test_mower_1_sync_clock"
+    await setup_integration(hass, mock_config_entry)
+    state = hass.states.get(entity_id)
+    assert state.name == "Test Mower 1 Sync clock"
+
+    values = mower_list_to_dictionary_dataclass(
+        load_json_value_fixture("mower.json", DOMAIN)
+    )
+    mock_automower_client.get_status.return_value = values
+
+    await hass.services.async_call(
+        domain="button",
+        service=SERVICE_PRESS,
+        target={ATTR_ENTITY_ID: entity_id},
+        blocking=True,
+    )
+    mocked_method = getattr(mock_automower_client.commands, "set_datetime")
+    # datetime(2024, 2, 29, 11, tzinfo=datetime.UTC) is in local time of the tests
+    # datetime(2024, 2, 29, 12, tzinfo=zoneinfo.ZoneInfo(key='Europe/Berlin'))
+    mocked_method.assert_called_once_with(
+        TEST_MOWER_ID,
+        datetime.datetime(2024, 2, 29, 12, tzinfo=zoneinfo.ZoneInfo("Europe/Berlin")),
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get(entity_id)
+    assert state.state == "2024-02-29T11:00:00+00:00"
+    getattr(mock_automower_client.commands, "set_datetime").side_effect = ApiException(
```

Comment:
idem

Reply:
done in https://github.com/home-assistant/core/pull/125689/commits/ef7a14c3a8ffe3964f796a47d705b5bc5ba8b8d7